packages/@tinacms/graphql/src/builder/index.ts

```diff
@@ -741,50 +699,70 @@ export class Builder {
   public buildStaticDefinitions = () => staticDefinitions
 
   private _buildCollectionDocumentType = async (
-    collection: TinaCloudCollectionEnriched
+    collection: TinaCloudCollectionEnriched,
+    suffix: string = '',
+    extraFields: FieldDefinitionNode[] = [],
+    extraInterfaces: NamedTypeNode[] = []
   ) => {
     const documentTypeName = NAMER.documentTypeName(collection.namespace)
+    const templateInfo = this.tinaSchema.getTemplatesForCollectable(collection)
+    if (templateInfo.type === 'union') {
```

Comment:
Previously this wouldn't be a union even when a collection had `templates` because the union was at the `data` level. But now it's at the top. This might not seem like much of a change, but for collections with `templates` now you can't do something like this:

```graphql
# assume `post` has 2 templates, `blogPost` and `blockPost`
post(relativePath: $relativePath) {
  id
}
```
`id` is available, but only when disambiguating. Since both `blogPost` and `blockPost` implement the `Document` interface, we can get pretty close to the same behavior:
```graphql
{
  post(relativePath: $relativePath) {
    ...on Document {
      id
    }
  }
}
```
It's a decent workaround, but sort of problematic because if `post` was not a union, you would have been able to do that first example. So just something for us to be aware of

Reply:
I'm seeing an issue with this when trying to upgrade to the latest versions of `tinacms` and `@tinacms/cli` (since PR 2823 was released). The auto-generated GraphQL fails for all collections that have multiple `templates`. I'm seeing errors like the output below, as the `...on Document {` wrapper is not added to the generated queries. Has this been reported or should I create a GitHub issue to raise this @jeffsee55?

```
AggregateError: GraphQL Document Validation failed with 4 errors;
  Error 0: GraphQLDocumentError: Cannot query field "_sys" on type "Theme". Did you mean to use an inline fragment on "Document", "ThemePalette", or "ThemeSettings"?
    at /Users/Corey/project-name/.tina/__generated__/queries.gql:3:5

Error 1: GraphQLDocumentError: Cannot query field "id" on type "Theme". Did you mean to use an inline fragment on "Document", "Node", "ThemePalette", or "ThemeSettings"?
    at /Users/Corey/project-name/.tina/__generated__/queries.gql:11:5

Error 2: GraphQLDocumentError: Cannot query field "id" on type "Theme". Did you mean to use an inline fragment on "Document", "Node", "ThemePalette", or "ThemeSettings"?
    at /Users/Corey/project-name/.tina/__generated__/queries.gql:21:9

Error 3: GraphQLDocumentError: Cannot query field "_sys" on type "Theme". Did you mean to use an inline fragment on "Document", "ThemePalette", or "ThemeSettings"?
    at /Users/Corey/project-name/.tina/__generated__/queries.gql:22:9
    at Object.checkValidationErrors (/Users/Corey/project-name/node_modules/@graphql-tools/utils/index.js:1252:15)
    at codegen (/Users/Corey/project-name/node_modules/@graphql-codegen/core/index.js:175:15)
    at async generateTypes (/Users/Corey/project-name/node_modules/@tinacms/cli/dist/index.js:323:17)
    at async genTypes (/Users/Corey/project-name/node_modules/@tinacms/cli/dist/index.js:360:27)
    at async build2 (/Users/Corey/project-name/node_modules/@tinacms/cli/dist/index.js:1600:7)
    at async FSWatcher.<anonymous> (/Users/Corey/project-name/node_modules/@tinacms/cli/dist/index.js:1624:11)
	Typescript types => /Users/Corey/project-name/.tina/__generated__/types.ts
	GraphQL types ====> /Users/Corey/project-name/.tina/__generated__/schema.gql
```

The auto-generated query for the theme collection looks like this, which fails as there are multiple templates configured for that `theme` collection:
```
query theme($relativePath: String!) {
  theme(relativePath: $relativePath) {
    _sys {
      filename
      basename
      breadcrumbs
      path
      relativePath
      extension
    }
    id
    ...ThemeParts
  }
}
```

Reply:
**UPDATE:** I've posted this as an issue here: https://github.com/tinacms/tinacms/issues/2904, with an example repository to recreate the problem visible here: https://github.com/coreyaus/tinacms-issue-2904. Definitely happy to hear any ways I can help - I took a look at the code for the `@tinacms/cli` package but couldn't work out exactly how to start working on a fix or a failing test for this sorry!